core/templates/pages/about-page/about-page.controller.spec.ts

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for aboutPage.
+ */
+
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+describe('About Page', function() {
+  var $scope = null, ctrl = null;
+  var windowRef = new WindowRef();
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    $provide.value('WindowRef', windowRef);
+  }));
+  beforeEach(angular.mock.inject(function($injector) {
+    var $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    var directive = $injector.get('aboutPageDirective')[0];
+    ctrl = $injector.instantiate(directive.controller, {
+      $rootScope: $scope
+    });
+  }));
+
+  afterEach(function() {
+    // onhashchange and location.hash are reassigned because it shares
+    // same reference to the all test blocks, making the tests unpredictable.
+    windowRef.nativeWindow.onhashchange = null;
+    windowRef.nativeWindow.location.hash = '';
+  });
+
+  it('should click on about tab', function(done) {
+    // @ts-ignore
+    var addClassSpy = spyOn($.fn, 'addClass').and.callThrough();
+    // @ts-ignore
+    var removeClassSpy = spyOn($.fn, 'removeClass').and.callThrough();
+
+    ctrl.$onInit();
+    ctrl.onTabClick('about');
+
+    // setTimeout is being used here in order to wait onhashchange event to
+    // finish. setTimeout is executed only after call stack is empty.
+    // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop
+    setTimeout(function() {
+      expect(windowRef.nativeWindow.location.hash).toBe('#about');
+      // 2 calls for activeTab calls on onTabClick and 2 calls for
+      // activeTab calls on $onInit.
+      expect(addClassSpy).toHaveBeenCalledTimes(4);
```

Comment:
I think it's not possible to check the calls using `toHaveBeenCalledTimes` because this method evaluates the total times that the spy was called. However, I've done it in a different way.

Reply:
It's fine for the separate checks on addClassSpy etc. to compare against the cumulative totals, if the concern is that you can't "reset" the call counts between successive calls. I was just pointing out the need to have clear "before"/"after" conditions.

Reply:
Oh okay, I've done it!